### ISSUE TITLE
Excluding the schema from the generated SQL

### ISSUE THREAD

**Issue by jjsil** [knq/xo#55]
Context: we have several instances of the same database schema, under different schema names in a single MySQL installation. We would like to be able to switch between them just by changing the schema in the connection string we pass to our binary in a flag.

Problem: This doesn't seem possible with xo at the moment because it adds the schema to all table and proc names in the generated code. Switching between different schemas requires re-generating the xo code and re-compiling.

Possible solution?: New flag to disable schema concatenation in internal.schmafn. More than happy to send a pull request if this sounds reasonable.

Current: `UPDATE dev.User SET `
Desired: `UPDATE User SET `

**Comment by kenshaw** [comment]
There are a couple of approaches to accomplishing this, using the existing xo code:

1) Different packages for each schema. For example, `schema/v1` and `schema/v2`. You would then generate them, and use common interfaces for each. You could modify the existing template to easily (for one only) generate all the common interfaces (this is the approach I use with PG databases, and also why xo is very helpful, as it generates the same interfaces for the database).

The logic would then be something like:
```go
var schema CommonSchema
if environment == "v1" {
    schema, err = v1.NewV1Schema()
} else {
    schema, err = v2.NewV2Schema()
}
```

2) Using custom templates that prefix the schema name on the type's / func's generated. ie, instead of "func BooksByAuthorID()" you would modify the template to be "func {{ schemaName}}BooksByAuthorID()". This becomes super cumbersome / silly with large schemas, so likely it would be better to just do #1.

That said, I would absolutely love ANY pull request that is well written! If you feel the addition of something like what you described would be beneficial, absolutely submit it, and I'll give it a read through and try it out. The barriers here for getting your PR are fairly low -- it just needs to make sense to the project as whole and be well written!

**Comment by Xeoncross** [comment]
I agree that having the schema name concatenated to the table name is a problem when dealing with different testing/staging environments both manually and auto created. In fact, I'd love to see the schema name become optional instead of default (unless the default target audience is working across the multiple schemas using the same connection pool).
